tests/discovery/testrendezvous.nim

```diff
@@ -0,0 +1,176 @@
+{.used.}
+
+# Nim-Libp2p
+# Copyright (c) 2023 Status Research & Development GmbH
+# Licensed under either of
+#  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
+#  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+# at your option.
+# This file may not be copied, modified, or distributed except according to
+# those terms.
+
+import sequtils, strutils
+import chronos
+import ../../libp2p/[protocols/rendezvous, switch]
+import ../../libp2p/discovery/discoverymngr
+import ../helpers
+import ./utils
+
+suite "RendezVous":
+  teardown:
+    checkTrackers()
+
+  asyncTest "Request locally returns 0 for empty namespace":
+    let (nodes, rdvs) = setupNodes(1)
+    nodes.startAndDeferStop()
+
+    const namespaceEmpty = ""
+    let peerRecords = rdvs[0].requestLocally(namespaceEmpty)
+    check peerRecords.len == 0
+
+  asyncTest "Request locally returns registered peers":
+    let (nodes, rdvs) = setupNodes(1)
+    nodes.startAndDeferStop()
+
+    const namespace = "foo"
+    await rdvs[0].advertise(namespace)
+    let peerRecords = rdvs[0].requestLocally(namespace)
+
+    check:
+      peerRecords.len == 1
+      peerRecords[0] == nodes[0].peerInfo.signedPeerRecord.data
+
+  asyncTest "Unsubscribe Locally removes registered peer":
+    let (nodes, rdvs) = setupNodes(1)
+    nodes.startAndDeferStop()
+
+    const namespace = "foo"
+    await rdvs[0].advertise(namespace)
+    check rdvs[0].requestLocally(namespace).len == 1
+
+    rdvs[0].unsubscribeLocally(namespace)
+    let peerRecords = rdvs[0].requestLocally(namespace)
+    check peerRecords.len == 0
```

Comment:
similar check above didn't use variable few lines above; better avoid variable if it expire on the next line